File: C++/Minimum Absolute Difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        int min_diff=INT_MAX;
        vector<vector<int>> ans;
        
        
        for(int i=0;i<arr.size()-1;++i){
            int diff=0;
            diff=abs(arr[i+1]-arr[i]);
            if(min_diff>=diff){
                min_diff=diff;
            }
        }
        
        for(int i=0;i<arr.size()-1;++i){
            int diff=0;
            diff=abs(arr[i+1]-arr[i]);
            if(diff == min_diff){
                ans.push_back({arr[i],arr[i+1]});
            }
        }
        return ans;
    }
};

File: JS/Minimum Absolute Difference.js
/**
 * @param {number[]} arr
 * @return {number[][]}
 */
var minimumAbsDifference = function(arr) {
    arr.sort((a,b) => a-b);
    let min_diff=Number.MAX_VALUE;
    let ans=[];
    
    for(let i=0;i<arr.length;++i){
        let diff=0;
        diff=Math.abs(arr[i+1]-arr[i]);
        if(min_diff >= diff){
            min_diff=diff;
        }
    }
    
    for(let i=0;i<arr.length-1;++i){
        let diff=0;
        diff=Math.abs(arr[i+1]-arr[i]);
        if(diff == min_diff){
            ans.push([arr[i],arr[i+1]]);
        }
    }
return ans;
};